In [6]:
import random
from tqdm import tqdm

In [4]:
#teenie beanie babies simulation
#10 types of babies
#assume equal circulation
#find approximation for how many meals needed to get all 10

def beaniesimulation(Simcounter=5000000):
    results=0
    for i in range(Simcounter):
        beanietypes=set()
        counter=0
        while len(beanietypes)<10:
            counter+=1
            active=random.randint(1,10)
            beanietypes.add(active)
        results+=(counter)
    mean=results/Simcounter
    return mean


In [5]:
beaniesimulation()

29.2828216

Assuming that each meal had an equal chance of a specific teenie beanie, then according to the simulation one will have to buy around 29 meals. The simulation was conducted 5,000,000 times. Each simulation counted how long it took to get all 10 beanies. It then took the average amount and returned 29.282 times. This means on average one needs to buy 29 happy meals to get all 10 beanies. 

In [23]:
"""World series
Consists of 7 games max. First to 4 wins
first 2 at home, next 3 at yankee, and final 2 at home
there is home advantage where 55% chance of winning
how often will the team that starts at home win the series
"""
def Worldseriessim(Simcounter=9000000):
    with tqdm(total=Simcounter, desc="Processing...") as pbar:
        results=0
        for i in range(Simcounter):
            teamwins={'1':0, '2':0}
            for i in range(2):
                active = random.randint(1, 100)
                if active <= 55:
                    teamwins['1'] += 1
                else:
                    teamwins['2'] += 1
            
            for i in range(3):
                active = random.randint(1, 100)
                if active < 45:  
                    teamwins['1'] += 1
                else:
                    teamwins['2'] += 1
            
            for i in range(2):
                active = random.randint(1, 100)
                if active <= 55:
                    teamwins['1'] += 1
                else:
                    teamwins['2'] += 1

            if teamwins['1'] > teamwins['2']:
                results += 1
            
            pbar.update(1)
    mean=results/Simcounter
    return mean
mean=Worldseriessim()*100
print(f'The team which starts at home has a {mean}% chance of winning.')

Processing...: 100%|██████████| 9000000/9000000 [00:40<00:00, 224874.34it/s]

The team which starts at home has a 50.61324444444445% chance of winning.


The theoretical probability of eating a hot dorito for the first round of roulette is 1/6. 

In [27]:
def chiproulette(simcount=9000000):
    with tqdm(total=simcount, desc="Processing...") as pbar:
        results=0
        for i in range(simcount):
            if random.randint(1,6) == 6:
                results+=1
            pbar.update(1)
        final=(results/simcount)*100
    return final, simcount

pain, suffering=chiproulette()
print(f'After {suffering} amount of trials, we have found that on average you have a {pain}% chance of having the painful dorrito.')

Processing...: 100%|██████████| 9000000/9000000 [00:05<00:00, 1676901.86it/s]

After 9000000 amount of trials, we have found that on average you have a 16.660533333333333% chance of having the painful dorrito.


After running a significant amount of trials, the empirical probability should approach the theoretical probability due to the law of large numbers. My empirical probability is 16.6605% which is very close to the 16.6666% theoretical probability.

In [68]:
def roulettegame():
    playercount = int(input('How many players? '))
    players = {'player': [], 'number': []}
    
    print('Loading and spinning...')
    activechamber=random.randint(1,6)
    currentchamber=random.randint(1,6)
    for i in range(playercount):
        playername = input(f'What is the name of player {i + 1}? ')
        players['player'].append(playername)
        players['number'].append(i)

    print('Let’s see who goes first...')
    current_player_index = random.randint(0, playercount - 1) 
    current_player = players['player'][current_player_index]
    while len(players['player']) > 0:
        current_player = players['player'][current_player_index]
        print(f'{current_player} has the gun. Pull the trigger! (Press Enter)')
        input()
        
        if currentchamber==activechamber:
            print(f'BANG! {current_player} has a booboo!')
            players['player'].remove(current_player)
            if current_player_index >= len(players['player']):
                current_player_index = 0
            activechamber=random.randint(1,6)
            currentchamber=random.randint(1,6)

        else:
            print(f'Click! {current_player} survives!')
            current_player_index = (current_player_index + 1) % len(players['player'])            
            currentchamber+=1
            if currentchamber==7:
                currentchamber=1


In [ ]:
roulettegame()

In [52]:
def birthday(people=23, simulations=900000):
    with tqdm(total=simulations, desc="Processing...") as pbar:

        results=0
        currentsim=0
        for i in range(simulations):
            birthdaylist = []
            for i in range(people):
                birthday = random.randint(1, 365)
                if birthday in birthdaylist:
                    results += 1
                    break
                birthdaylist.append(birthday)
            pbar.update(1)
    fin=(results/simulations)*100
    return fin

In [53]:
fin=birthday()
print(fin)

Processing...: 100%|██████████| 900000/900000 [00:33<00:00, 26730.62it/s]

50.800222222222224
